custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+            var keys = Object.keys(items);
+            for (var i = 0, len = keys.length; i < len; i++) {
+              var item = items[keys[i]];
+              // if array or object, dive into it
+              if (keys[i] === "logo") {
+                target.logo = true;
+              }
+              if (item instanceof Object || item instanceof Array) {
+                nestedJsonldLookup(target, jsonldIds, item, depth++, context);
+              }
+            }
+          }
+        }
+
+        function addType(target, array, type, jsonld) {
+          link.href = type;
+          let www = false;
+          let hostname = link.hostname;
+          
+
+          if (hostname.startsWith("www.")) {
+            www = true;
+            hostname = hostname.substring("www.".length);
+          }
+
+          let name = hostname + link.pathname; 
+
+          let item = array.find(i => i.name === name);       
+          if (!item) {
+            item = {name: name, count: 0, jsonld: 0, microdata: 0, https: 0, http: 0, www: 0};
+            array.push(item);
+          }  
+
+          item.count++;
+
+          if (link.protocol === 'https:') 
+            item.https++;        
+          else 
+            item.http++;
+
+          if (www) 
+            item.www++;   
+
+          if (jsonld) 
+            item.jsonld++;
+          else 
+            item.microdata++;
+
+          if (!target.context_hostnames.includes(hostname)) {
+            target.context_hostnames.push(hostname);
+          }
+
+          if (name === "schema.org/SearchAction") {
+            target.sitelinks_search_box = true;
+          }
+        }
+
+        function gatherStructuredData(d) {
+
+          let target = {
+            jsonld_and_microdata_types: [],
+            jsonldIds: 0,
+            jsonldReferencedIds: 0,
+            microdataIds: 0,
+            microdataReferencedIds: 0,
+            jsonlsMicrodataCommonIds: 0,
+            logo: false,
+            sitelinks_search_box: false,
+            same_as_values: [],
+            items_by_format: {
+              microformats2: 0,
+              microdata: 0,
+              jsonld: 0,
+              rdfa: 0
+            },
+            context_hostnames: []
+          };
+
+          let jsonldIds = {};
+
+          // json-ld
+          let jsonld_scripts = Array.from(d.querySelectorAll('script[type="application/ld+json"]'));
+
+          target.jsonld_scripts = {
+            count: jsonld_scripts.length,
+            errors: jsonld_scripts.filter(e => {
+              try {
+                var cleanText = e.textContent.trim();
+                var cleanText = cleanText.replace(/^\/\*(.*?)\*\//g, ''); // remove * comment from start (could be for CDATA section) does not deal with multi line comments
+                var cleanText = cleanText.replace(/\/\*(.*?)\*\/$/g, ''); // remove * comment from end (could be for CDATA section) does not deal with multi line comments
```

Comment:
Not clear if this repetition was intentional. I suspect not since the variable is declared three times as well.

Reply:
They are slightly different. The extra vars should not be there. Good spot. Updating.